qiskit_experiments/library/tomography/basis/cache_method.py

```diff
@@ -0,0 +1,75 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Method decorator for caching regular methods in class instances.
+"""
+
+from typing import Union, Dict, Callable
+import functools
+
+
+def cache_method(cache: Union[Dict, str] = "_cache") -> Callable:
+    """Decorator for caching class instance methods.
+
+    Args:
+        cache: The cache or cache attribute name to use. If a dict it will
+               be used directly, if a str a cache dict will be created under
+               that attribute name if one is not already present.
```

Comment:
Do you think this argument will be used in practise? Having trouble thinking when I would want something else to own the cache.

Also wondering if the cache name could be tied to the type to avoid having to specify the same cache name for every cached method within a class:

```python
...
    def _get_cache(self):
        cache_name = getattr(type(self), "_CACHE_NAME", "_cache")
        if not hasattr(self, cache_name):
            setattr(self, cache_name, {})
        return getattr(self, cache_name)
...

# default cache name
class Foo:
    @cache_method
    def foo(self, x):
        return x

# custom cache name
class Foo2:
    _CACHE_NAME = "_another_cache_name"
    @cache_method
    def foo(self, x):
        return x
```

Reply:
I originally added it because we planned on doing more general caching in experiments, but that idea was tabled awhile ago so its not strictly needed anymore for use here in the tomography basis classes. So I can remove it